src/Mapper/DoctrineCollector.php

```diff
@@ -95,12 +95,22 @@ public function addDiscriminator(string $class, string $key, string $discriminat
     public function addDiscriminatorColumn(string $class, ColumnDefinitionBuilder $columnDef): void
     {
         if (!isset($this->discriminatorColumns[$class])) {
-            $this->discriminatorColumns[$class] = $columnDef;
+            $this->discriminatorColumns[$class] = $columnDef->getOptions();
         }
     }
 
-    public function addInheritanceType(string $class, string $type): void
+    /**
+     * @param int $type
+     */
+    public function addInheritanceType(string $class, $type): void
     {
+        // NEXT_MAJOR: Move int check to method signature
+        if (!\is_int($type)) {
+            throw new \InvalidArgumentException(
+                'Argument 2 is not an integer, expecting an integer'
```

Comment:
The argument declaration will produce an exception of type `\TypeError`, with a message similar to:
```suggestion
            throw new \TypeError(
                \sprintf('Argument 2 passed to %s() must be of the type int, %s given.', __METHOD__, is_object($type) ? get_class($type) : gettype($type));
```
The same applies for other exceptions added in this PR.

Reply:
Fixed